crates/core/tedge_agent/src/restart_manager/builder.rs

```diff
@@ -0,0 +1,77 @@
+use crate::restart_manager::actor::RestartManagerActor;
+use crate::restart_manager::actor::RestartManagerConfig;
+use crate::state_repository::actor::StateRepositoryAction;
+use crate::state_repository::actor::StateResult;
+use tedge_actors::futures::channel::mpsc;
+use tedge_actors::Builder;
+use tedge_actors::ClientMessageBox;
+use tedge_actors::DynSender;
+use tedge_actors::LinkError;
+use tedge_actors::LoggingReceiver;
+use tedge_actors::NoConfig;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::RuntimeRequestSink;
+use tedge_actors::ServiceProvider;
+use tedge_api::RestartOperationRequest;
+use tedge_api::RestartOperationResponse;
+
+pub struct RestartManagerBuilder {
+    config: RestartManagerConfig,
+    input_receiver: LoggingReceiver<RestartOperationRequest>,
+    converter_sender: DynSender<RestartOperationResponse>,
+    state_message_box: ClientMessageBox<StateRepositoryAction, StateResult>,
+    signal_sender: mpsc::Sender<RuntimeRequest>,
+}
+
+impl RestartManagerBuilder {
+    pub fn new(
+        config: RestartManagerConfig,
+        converter_actor: &mut impl ServiceProvider<
+            RestartOperationResponse,
+            RestartOperationRequest,
+            NoConfig,
+        >,
```

Comment:
On a first look, I was surprised to see the request and response arguments flipped, but it finally made sense while thinking of the `RestartManagerBuilder` as a `ServiceConsumer` that receives `RestartOperationRequest` and sends back `RestartOperationResponse`.